community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/internal/concurrent/SessionWorker.java

```diff
@@ -81,6 +81,11 @@ public void run()
                     {
                         executeBatch( batch );
                     }
+
+                    if(workQueue.isEmpty())
```

Comment:
Oh, lol. That's a completely unrelated thing :p I was exploring moving the logic of when to flush outbound buffers to be when then queue's empty here. Will remove